framework/src/interfacekernels/InterfaceKernelBase.C

```diff
@@ -63,10 +64,14 @@ InterfaceKernelBase::validParams()
       "save-in jacobian contributions from the primary ('p') or secondary side ('s').");
 
   // InterfaceKernels always need one layer of ghosting.
-  params.addRelationshipManager("ElementSideNeighborLayers",
-                                Moose::RelationshipManagerType::GEOMETRIC |
-                                    Moose::RelationshipManagerType::ALGEBRAIC |
-                                    Moose::RelationshipManagerType::COUPLING);
+  params.addRelationshipManager(
+      "ElementSideNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC | Moose::RelationshipManagerType::ALGEBRAIC |
+          Moose::RelationshipManagerType::COUPLING,
+      [](const InputParameters & obj_params, InputParameters & rm_params) {
+        rm_params.set<bool>("create_full_coupling_matrix") =
+            obj_params.get<bool>("is_ad") && Moose::globalADIndexing();
+      });
```

Comment:
Could we default to doing this with global indexing and using AD?  And then provide a simple way for the user to override/disable the full coupling if they don't want it?

Reply:
This PR does already provide the means to override full coupling. I think you're right that we can just default to full coupling when `_fe_problem.haveADObjects()`. It's an excellent question how I settled on this method when what you proposed is soooo much simpler and does the exact same thing. A rabbit hole of relationship manager thinking I guess :laughing: Anyway yea I'll remove all of this logic. And then there will still be some good RM changes that come out of this PR anyways

Reply:
See https://github.com/idaholab/moose/pull/16397#issuecomment-747764789

Reply:
This kind of thing should only be done at the framework level in core residual/Jacobian computing classes. Application developers cannot add new residual/Jacobian objects that do not inherit from the core framework residual/Jacobian classes. The only time someone would have to reproduce this syntax that I've added is if they're adding some new core residual/Jacobian computing system. 

Because the problem initialization happens before we construct our residual objects, our solution essentially *has* to come out of `(Derived)ResidualObject::validParams`, unless you have a more clever idea. I do like the idea of toggling a change in variable coupling through (coupling) `RelationshipManagers/GhostingFunctors` since this is conceptually what they are supposed to handle: controlling construction of the sparsity pattern.